scipy/stats/stats.py

```diff
@@ -3543,19 +3547,22 @@ def count_rank_tie(ranks):
     x, y = x[perm], y[perm]
     x = np.r_[True, x[1:] != x[:-1]].cumsum(dtype=np.intp)
 
-    con, dis = _kendall_condis(x, y)  # concordant & discordant pairs
+    dis = _kendall_dis(x, y)  # discordant pairs
 
     obs = np.r_[True, (x[1:] != x[:-1]) | (y[1:] != y[:-1]), True]
     cnt = np.diff(np.where(obs)[0]).astype('int64', copy=False)
 
     ntie = (cnt * (cnt - 1) // 2).sum()  # joint ties
-    xtie = count_rank_tie(x) - ntie      # ties only in x
-    ytie = count_rank_tie(y) - ntie      # ties only in y
+    xtie = count_rank_tie(x)             # ties in x
+    ytie = count_rank_tie(y)             # ties in y
+
+    tot = (size * (size - 1)) // 2
 
-    if con + dis + xtie == 0 or con + dis + ytie == 0:
+    if xtie == tot or ytie == tot:
         return KendalltauResult(np.nan, np.nan)
 
-    tau = (con - dis) / np.sqrt(con + dis + xtie) / np.sqrt(con + dis + ytie)
+    # Limit range to fix computational errors
+    tau = min(1., max(-1., (tot - xtie - ytie + ntie - 2 * dis) / np.sqrt(tot - xtie) / np.sqrt(tot - ytie)))
```

Comment:
Please keep the line lengths <80 chars. Here I'd separate clipping, `min(1, max(-1, ...))` from computation.
Also, a comment that the numerator is actually `con - dis` would be nice.